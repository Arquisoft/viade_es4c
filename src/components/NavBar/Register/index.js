import RegisterModal from "./register-modal.component";

export default RegisterModal;