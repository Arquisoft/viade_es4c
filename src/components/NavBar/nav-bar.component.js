import React from "react";
import {Image, Navbar} from "react-bootstrap";
import Nav from "react-bootstrap/Nav";
import LoginComponent from "./Login";
import RegisterContainer from "./Register";
import {navbar} from "../../contexts";
import {CustomModal} from "../index";

/**
 * Navigation bar which contains de actions of a user not logged in
 *  @returns {*}
 */
const NavBar = () => {

	return (
		<Navbar bg="light" expand="lg">
			<Navbar.Brand href={process.env.PUBLIC_URL + navbar.logo.href}>
				<Image src={process.env.PUBLIC_URL + navbar.logo.src} alt={navbar.logo.alt}/>
			</Navbar.Brand>
			<Navbar.Toggle aria-controls="basic-navbar-nav"/>
			<Navbar.Collapse id="basic-navbar-nav">
				<Nav className="mr-auto">
				</Nav>
				<Nav>
					{/* Button that opens the login modal */}
					<CustomModal img="/img/buttons/login.png" text="Login"
								component={<LoginComponent/>}/>
					{/* Button that opens the register modal */}
					<CustomModal img="/img/buttons/register.png" text="Register"
								component={<RegisterContainer/>}/>
				</Nav>
			</Navbar.Collapse>
		</Navbar>
	);
};

export default NavBar;
