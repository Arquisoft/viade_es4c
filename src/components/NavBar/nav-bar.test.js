import React from 'react';
import { shallow } from 'enzyme';
import NavBar from "./nav-bar.component";

describe('NavBar', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<NavBar debug />);
  
    expect(component).toMatchSnapshot();
  });
});
