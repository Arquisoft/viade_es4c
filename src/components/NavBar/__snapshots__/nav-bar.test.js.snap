// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly in "debug" mode 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  variant="light"
>
  <NavbarBrand
    href=""
  >
    <Image
      alt="Viade"
      fluid={false}
      rounded={false}
      roundedCircle={false}
      src="/img/logo.svg"
      thumbnail={false}
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="basic-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    />
    <Nav
      fill={false}
      justify={false}
    >
      <LoginModal />
      <LoginModal />
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
